---
title: "February 2024, Daily Log"
author: "Kathleen Durkin"
date: "last-modified"
categories: ["daily log"]
format:
  html:
    toc: true
---

## 29 February 2024

#### Thursday

## 28 February 2024

#### Wednesday

## 27 February 2024

#### Tuesday

## 26 February 2024

#### Monday

## 23 February 2024

#### Friday

## 22 February 2024

#### Thursday

## 21 February 2024

#### Wednesday

## 20 February 2024

#### Tuesday

## 19 February 2024

#### Monday

## 16 February 2024

#### Friday

## 15 February 2024

#### Thursday

-   reading -- I think I have a direction I want to go for the oyster project! Also looks like there's a gap in the literature

-   ran idea by Ariana and she had some super helpful feedback about what variables need to be considered

## 14 February 2024

#### Wednesday

-   reading more oyster papers

    -   triploidy inhibits gametogenisis -- does it also affect oysters' hermaphrodism? I.e. are triploid oysters less able to switch between male/female? If gamete production is inhibited, can you even tell?

## 13 February 2024

#### Tuesday

-   Class -- really interesting discussion of this set of critique/comment/letter to the editor papers basically arguing about the Gill-Oxygen Limitation Theory

-   Started high-level lit review of oysters to try to figure out what question(s) I want to ask for an oyster experiment. Brianik & Allem 2023 review suggests empirical studies on oyster triploidy could be a really relevant area

## 12 February 2024

#### Monday

-   Classwork and classes

## 09 February 2024

#### Friday

-   E5 meetings

-   trying to figure out what exactly the --ad-all-nodes option does in MirMachine, the documentation is p vague

-   readings/synthesis post for fish510 on monday (**5** papers this weekðŸ¥´)

## 08 February 2024

#### Thursday

-   note to self: go back and add references in my notebook posts

-   looking into how to apply WGCNA to mulitomics datasets, since documentation seems to be focusd more on applying to single-omics with multiple treatments? Found som relevant papers, though I'm still a little confused by methods...

-   \^ also found different toolkit called [mixOmics](https://mixomicsteam.github.io/mixOmics-Vignette/id_01.html), specifically intended for multi-omics analysis, but all of the visualization aptions look p different from WGCNA standard visualizations and I'm not that familiar with some of the terminology, so it isn't clear to me upon first read through how it compares to WGCNA approach...

## 07 February 2024

#### Wednesday

-   read/watched some stuff on using WGCNA (for E5 RNAseq and sRNAseq data)

-   Thought I was done w P. evermanni, but Sam pointed out that gff and bed files handle genomic coordinates differently and that, even though bedtools getfasta lists gffs as acceptable input formats, probably still treats them the same as bed files, so all of my transcriptome fastas could be off by 1-2 sequences. Probably doesn't affect end result of kallisto much, since it's pseudoaligning full reads, but still need to fix it. Should be a straightforward fix, the biggest pain will be waiting another 4 hours for the adjusted transcriptome generation code to run againðŸ˜­

-   Did a quick and dirty [test](/posts/projects/misc/2024_02_08_bedtools_gffvbed.qmd) using both a gff and a bed file as input to bedtools getfasta and both resulted in the same output, suggesting bedtools does appropriately differentiate between the formats! I'll still convert to bed before generating the P.evermanni transcriptome though, just in case

## 06 February 2024

#### Tuesday

-   Wrote some checks for the P.evermanni transcriptome generation script, and I'm pretty confident now that it's running correctly. I did a final rerun of the generation and wrote a [notebook post](/posts/projects/E5_coral/2024_02_06_RNAseq_kallisto.qmd) for the kallisto process for all three deep dive species. Will need to update it tomorrow once I have the final P.evermanni transcriptome and can run it through kallisto as well.

-   510 class

## 05 February 2024

#### Monday

-   Didn't allocate any time last week for the two class assignments I have due today, so cranking those out (proposal introduction for cod project, read and write up discussion for seminar papers)
-   Proposal writing class
-   Checked output for my P.evermanni genome fasta script, then generalized the script (i.e. changed input/output file assignments to use the bash variables I set at the beginning of the document) and reran. It ran much quicker this time around and resulted in a much smaller output file, which means one of the two times I ran the script something was wrong. I did some brief poking around in the two outputs and it looks like maybe the original run was accidentally done using the full gff file, instead of the CDS-only gff? I don't have time to resolve it tonight, but I'll check again tomorrow and run the code and output by someone (Zach, Steven)

## 02 February 2024

#### Friday

-   E5 molecular group meeting

-   Wrote a script to appropriately extract, group, and concatenate CDS fastas to build a genome fasta file for *P.evermanni* kallisto transcript abundance quantification (see details and some pseudocode [here](/posts/projects/E5_coral/2024_01_31_Peve_kallisto.qmd)). Script is pretty slow so I'll have to wait till Monday to get and check output file.

## 01 February 2024

#### Thursday

-   Lab meeting -- Ariana led a really great discussion about project planning and shared a *super* helpful sample [document](https://docs.google.com/spreadsheets/d/11Xw5QXnjyyJYEDEpGr4dzZ3VDq9JwmtPsxUFJBsVlW8/edit#gid=0) of how that might play out for a project, including timelines

-   Reviewed January goals, set February goals, and set up this month's daily log

-   added some pseudocode to the [Peve kallisto troubleshooting post](/posts/projects/E5_coral/2024_01_31_Peve_kallisto.qmd) and began implementing
