---
title: "November 2023, Daily Log"
author: "Kathleen Durkin"
date: "2023-11"
categories: ["daily log"]
format:
  html:
    toc: true
---

# Daily Log

## 30 November 2023

#### Thursday

## 29 November 2023

#### Wednesday

## 28 November 2023

#### Tuesday

## 27 November 2023

#### Monday

Another paper+questions.

[Gajigan et al. 2017](https://doi.org/10.1111/mec.14130):

-   Many of the miRNAs identified in *A.digitifera* are novel (not identified in any other cnidarians or other taxa) -- are miRNAs generally this divergent accross closely related taxa, or is this proportion of novel miRNAs unusual?

## 24 November 2023

#### Friday

-   On Vacation

## 23 November 2023

#### Thursday

-   On Vacation
-   Happy Birthday to me 🎉

## 22 November 2023

#### Wednesday

-   On Vacation

## 21 November 2023

#### Tuesday

-   On Vacation

## 20 November 2023

#### Monday

Read some papers on ncRNAs, questions for next Science Hour/meeting with Steven below:

[Zhu et al. 2022](https://doi.org/10.1016/j.ygeno.2022.110404): I thought this paper was *sooooo* cool! The possibility of cross-kingdom miRNA effects is *crazy*

-   What is KEEG? (referenced similarly to GO) Just a different database?

-   "Rho-related" terms? (In GO context)

-   No idea what Fig. 1 is showing

-   I've noticed that GO annotations I've seen for corals never map to known *coral* functions -- they're always to similar things in other more-studied taxa (e.g. insects, nematodes). How often are GO annotations originally identified/set in one taxa validated for application in other taxa? How would that work? Is it even necessary?

-   So Zhu et al. 2022 found symbiont-associated miRNAs in coral transcriptome, but also mentioned in intro that (in other studies) foreign miRNA presence was correlated with intake (e.g. when human ate certain plant associated miRNAs present in human serum, but would disappear if they stopped eating that plant). Does that mean that presence/function of symbiont miRNAs in corals change when the coral is bleached (aka, the symbionts are gone)? If some symbiont miRNAs may be affecting coral immune response, could this play into coral immunocompromisation during bleaching?

[Trigg et al. 2021](https://doi.org/10.1111/1755-0998.13542)

Also, not from either of the papers, but I was kind of confused about Sam's discussion of trying different trimming sizes to ID miRNAs. I've only taken a basic molgen class though, which didn't go very in-depth in to how RNAs (especially ncRNAs) work, so this may be a basic knowledge gap. In contrast to DNA, where all of the different introns/exons, etc. are contained in one long strand of DNA, I thought transcribed RNAs were separate molecules. If that's the case my understanding on how we sequence miRNAs is that we isolate all the RNAs of a very small size (20-30bp) and then sequence them -- why, then, would trimming be necessary? And how do you decide to do trimming? Do all miRNAs have some identical feature that indicates the beginning/end of them (like start/stop codons in DNA?

## 17 November 2023

#### Friday

-   Met with E5 molecular working group, reviewed status of long and short ncRNA investigations

-   Attended GSS department presentations

-   Presented at GSS!

## 16 November 2023

#### Thursday

-   Met with Steven to go over number and location of cod samples we want to send for RNAseq. Decided we probably want to send 160 samples total, 80 liver samples and 80 gill. Within each, 5 from each tank (4 tanks per temp treatment), for a total of 20 samples from each of the 4 temperature treatments.

## 15 November 2023

#### Wednesday

-   Revised slides and practiced for GSS presentation (doing rehearsal presentation in lab meeting tomorrow)

-   Started working on some sRNA analysis for *P.evermanni*, but made the classic blunder of failing to pull from Github before I started, so I realized like an hour in that Sam had already finished the trimming and analysis I was working on🤦‍♀️

## 14 November 2023

#### Tuesday

I think I resolved my issues pushing from Raven to github! I backed up all my work to Gannet, then undid all my local commits, and recommitted/pushed only the .gitignore file. Once it was pushed, I reran my code to generate all the outputs again, and was able to successfully commit and push. I'm guessing the first time I tried to push stuff to Github, since I hadn't preemptively pushed the gitignore, it tried to push the too-large files that should have been excluded.

-   Modified Sam's code for *P.meandrina* sRNA [trimming](https://github.com/urol-e5/deep-dive/blob/main/F-Pmea/code/08-Pmea-sRNAseq-trimming.md), [miRTrace](https://github.com/urol-e5/deep-dive/blob/main/F-Pmea/code/09-Pmea-sRNAseq-miRTrace.md), and [BLAST](https://github.com/urol-e5/deep-dive/blob/main/F-Pmea/code/10-Pmea-sRNAseq-BLASTn.md).

## 13 November 2023

#### Monday

## 10 November 2023

#### Friday

-   Read some miRNA papers (Liew et al. 2014)

## 09 November 2023

#### Thursday -- Lab Meeting, Debugging Raven-\>Github Issue

-   Lab Meeting! DEI discussion

-   Talked with Steven about Raven-\>Github issue (can't currently push from Raven). Meanwhile, backed up current work to Gannet

-   Classes and assignments

## 08 November 2023

#### Wednesday -- E5 sRNA work (*P. meandrina*) and Website Reformatting

-   Debugging my modification of Sam's sRNA trimming code for application to *P. meandrina.*

-   Reformatting Lab Notebook structure to be based on monthly "Daily Log" posts with entries for every day of the month, rather than creating individual posts for every single lab notebook entry! (thanks to Chris for the idea😄)

-   Read (aka listened to the audiobook of) Chpt.1 of "Race After Technology" for lab DEI discussion tomorrow

## 07 November 2023

#### Tuesday -- Classes and Homework

Mostly classes, seminar readings, and QSCI assignment today.

## 06 November 2023 

#### Monday -- E5 Molecular Working Group

Met with the E5 molecular working group to review status of long and short ncRNA deep dive. Got started on applying Sam's code for sRNA analysis to *P. meandrina*.

Also presented a version of my GSS talk (UCE target enrichment to improve soft coral species delimitation) for FISH510 assignment and got some helpful feedback! Will make some tweaks and cut down time before my practice talk in Nov. 16 lab meeting.

## 03 November 2023

#### Friday

Set November monthly goals

Met with Steven during Science hour

## 02 November 2023

#### Thursday -- Calculating RNASeq sample size

Calculated necessary sample size for cod RNA sequencing, given an assumed coefficient of variation and detectable difference in expression -- view document [here](https://github.com/RobertsLab/project-cod-temperature/blob/main/code/04-RNASeq-sample-size.md).

Met with the E5 molecular working group for the first time!

## 01 November 2023

#### Wednesday -- SICB Registration, Reviewing October Goals

In the spirit of yesterday's holiday, I completed the scary task of registering for SICB out-of-pocket and then submitting the reimbursement request 🥴. This afternoon I also attended a workshop on making research presentations and slides more communicative and accessible, led by Tim.

With the start of a new month I reviewed my October goals (see that here)

Also, as seen above, I've figured out how to add emojis to my lab notebook -- gamechanger!
