---
title: "sRNA expression summaries"
author: "Kathleen Durkin"
date: "2024-11-11"
categories: ["E5-coral"]
format:
  html:
    toc: true
engine: knitr
bibliography: ../../../references.bib
---

I summarized sRNA expression for all three E5 species. This includes summary of both raw and normalized counts of all sRNAs, miRNAs, and siRNAs. sRNA, miRNA, and siRNA identification performed using ShortStack 4.1.0.

[`deep-dive-expression` repo](https://github.com/urol-e5/deep-dive-expression/tree/main)

Rendered code files:

-   [Apul](https://github.com/urol-e5/deep-dive-expression/blob/main/D-Apul/code/03.1-Apul-sRNA-summary.md)

-   [Peve](https://github.com/urol-e5/deep-dive-expression/blob/main/E-Peve/code/03.1-Peve-sRNA-summary.md)

-   [Ptuh](https://github.com/urol-e5/deep-dive-expression/blob/main/F-Ptuh/code/03.1-Ptuh-sRNA-summary.md)

This also included generating count matrices for raw, normalized, and variance stabilized counts of all sRNA, just miRNA, and just siRNA.

count matrices:

-   [Apul](https://github.com/urol-e5/deep-dive-expression/tree/main/D-Apul/output/03.1-Apul-sRNA-summary)

-   [Peve](https://github.com/urol-e5/deep-dive-expression/tree/main/E-Peve/output/03.1-Peve-sRNA-summary)

-   [Ptuh](https://github.com/urol-e5/deep-dive-expression/tree/main/F-Ptuh/output/03.1-Ptuh-sRNA-summary)

As a reminder, normalization and variance stabilization are two different transformations that serve different purposes.

**Normalizing** the transcript counts **adjusts for differences in library size** or sequencing depth, but **retains count-like properties**. Normalized counts are most useful for things like visualizing expression levels and differential expression analysis.

**Variance stabilization** aims to make the variance of the transformed data approximately independent of the mean, **reducing heteroscedasticity** (the relationship between variance and mean) and "smoothing" out the variance at low counts. Notably, the transformed data is **no longer on the original count scale**. The transformation makes the variance roughly constant across the range of counts, which makes it easier to interpret patterns in the data visually. Variance stabilized data is most useful for exploratory data analysis, like PCA, clustering, and heatmaps, and is also the transformation we'll want to use before WGCNA.

Here's some of the figures generated in these summary docs

## Total transcripts (raw)

All sRNA:

::: {layout-ncol="3"}
![Apul](https://github.com/urol-e5/deep-dive-expression/raw/main/D-Apul/code/03.1-Apul-sRNA-summary_files/figure-gfm/transcript-counts-plot-1.png){width="400"}

![Peve](https://github.com/urol-e5/deep-dive-expression/raw/main/E-Peve/code/03.1-Peve-sRNA-summary_files/figure-gfm/transcript-counts-plot-1.png){width="400"}

![Ptuh](https://github.com/urol-e5/deep-dive-expression/raw/main/F-Ptuh/code/03.1-Ptuh-sRNA-summary_files/figure-gfm/transcript-counts-plot-1.png){width="400"}
:::

miRNA:

::: {layout-ncol="3"}
![Apul](https://github.com/urol-e5/deep-dive-expression/raw/main/D-Apul/code/03.1-Apul-sRNA-summary_files/figure-gfm/miRNA-counts-plot-1.png)

![Peve](https://github.com/urol-e5/deep-dive-expression/raw/main/E-Peve/code/03.1-Peve-sRNA-summary_files/figure-gfm/miRNA-counts-plot-1.png)

![Ptuh](https://github.com/urol-e5/deep-dive-expression/raw/main/F-Ptuh/code/03.1-Ptuh-sRNA-summary_files/figure-gfm/miRNA-counts-plot-1.png)
:::

siRNA:

::: {layout-ncol="3"}
![Peve](https://github.com/urol-e5/deep-dive-expression/raw/main/D-Apul/code/03.1-Apul-sRNA-summary_files/figure-gfm/siRNA-counts-plot-1.png)

![Peve](https://github.com/urol-e5/deep-dive-expression/raw/main/E-Peve/code/03.1-Peve-sRNA-summary_files/figure-gfm/siRNA-counts-plot-1.png)

![Ptuh](https://github.com/urol-e5/deep-dive-expression/raw/main/F-Ptuh/code/03.1-Ptuh-sRNA-summary_files/figure-gfm/siRNA-counts-plot-1.png)
:::

## Total transcripts (normalized)

All sRNA:

::: {layout-ncol="3"}
![Apul](https://github.com/urol-e5/deep-dive-expression/raw/main/D-Apul/code/03.1-Apul-sRNA-summary_files/figure-gfm/norm-transcript-counts-plot-1.png)

![Peve](https://github.com/urol-e5/deep-dive-expression/raw/main/E-Peve/code/03.1-Peve-sRNA-summary_files/figure-gfm/norm-transcript-counts-plot-1.png)

![Ptuh](https://github.com/urol-e5/deep-dive-expression/raw/main/F-Ptuh/code/03.1-Ptuh-sRNA-summary_files/figure-gfm/norm-transcript-counts-plot-1.png)
:::

miRNA:

::: {layout-ncol="3"}
![Apul](https://github.com/urol-e5/deep-dive-expression/raw/main/D-Apul/code/03.1-Apul-sRNA-summary_files/figure-gfm/miRNA-norm-transcript-counts-plot-1.png)

![Peve](https://github.com/urol-e5/deep-dive-expression/raw/main/E-Peve/code/03.1-Peve-sRNA-summary_files/figure-gfm/miRNA-norm-transcript-counts-plot-1.png)

![Ptuh](https://github.com/urol-e5/deep-dive-expression/raw/main/F-Ptuh/code/03.1-Ptuh-sRNA-summary_files/figure-gfm/miRNA-norm-transcript-counts-plot-1.png)
:::

siRNA:

::: {layout-ncol="3"}
![Apul](https://github.com/urol-e5/deep-dive-expression/raw/main/D-Apul/code/03.1-Apul-sRNA-summary_files/figure-gfm/siRNA-norm-transcript-counts-plot-1.png)

![Peve](https://github.com/urol-e5/deep-dive-expression/raw/main/E-Peve/code/03.1-Peve-sRNA-summary_files/figure-gfm/siRNA-norm-transcript-counts-plot-1.png)

![Ptuh](https://github.com/urol-e5/deep-dive-expression/raw/main/F-Ptuh/code/03.1-Ptuh-sRNA-summary_files/figure-gfm/siRNA-norm-transcript-counts-plot-1.png)
:::

## Unique transcripts

All sRNA:

Little variation within a species, but quite a large difference among species. Apul specimens expressed \~19,000 unique sRNA transcripts, Peve expressed \~15,700, and Ptuh specimens expressed dramatically fewer at \~6700 unique sRNA transcripts.

::: {layout-ncol="3"}
![Apul](https://github.com/urol-e5/deep-dive-expression/raw/main/D-Apul/code/03.1-Apul-sRNA-summary_files/figure-gfm/total-unique-transcripts-plot-1.png)

![Peve](https://github.com/urol-e5/deep-dive-expression/raw/main/E-Peve/code/03.1-Peve-sRNA-summary_files/figure-gfm/total-unique-transcripts-plot-1.png)

![Ptuh](https://github.com/urol-e5/deep-dive-expression/raw/main/F-Ptuh/code/03.1-Ptuh-sRNA-summary_files/figure-gfm/total-unique-transcripts-plot-1.png)
:::

miRNA:

Despite the differences in total unique sRNA transcripts, all three species have **similar numbers of unique miRNAs** (though, note our deep-dive exploration showed there is little miRNA overlap among species).

::: {layout-ncol="3"}
![Apul](https://github.com/urol-e5/deep-dive-expression/raw/main/D-Apul/code/03.1-Apul-sRNA-summary_files/figure-gfm/total-unique-miRNA-plot-1.png)

![Peve](https://github.com/urol-e5/deep-dive-expression/raw/main/E-Peve/code/03.1-Peve-sRNA-summary_files/figure-gfm/total-unique-miRNA-plot-1.png)

![Ptuh](https://github.com/urol-e5/deep-dive-expression/raw/main/F-Ptuh/code/03.1-Ptuh-sRNA-summary_files/figure-gfm/total-unique-miRNA-plot-1.png)
:::

siRNA:

There is again a more noticable difference in numbers of unique siRNAs. **Apul have \~120, Peve \~155, and Ptuh \~90**.

::: {layout-ncol="3"}
![Apul](https://github.com/urol-e5/deep-dive-expression/raw/main/D-Apul/code/03.1-Apul-sRNA-summary_files/figure-gfm/total-unique-siRNA-plot-1.png)

![Peve](https://github.com/urol-e5/deep-dive-expression/raw/main/E-Peve/code/03.1-Peve-sRNA-summary_files/figure-gfm/total-unique-siRNA-plot-1.png)

![Ptuh](https://github.com/urol-e5/deep-dive-expression/raw/main/F-Ptuh/code/03.1-Ptuh-sRNA-summary_files/figure-gfm/total-unique-siRNA-plot-1.png)
:::
